ui/src/hooks/useFileManage.ts

```diff
@@ -537,45 +553,60 @@ export const useFileManage = () => {
           onError: () => void;
           onProgress: (e: { percent: number }) => void;
         }) => {
-          handleFileUpload(<WebSocket>socket, fileList, onProgress, onFinish, onError);
+          if (socket) {
+            handleFileUpload(socket, fileList, onProgress, onFinish, onError);
+          }
         }
       );
 
       mittBus.on('download-file', ({ path, is_dir }: { path: string; is_dir: boolean }) => {
-        handleFileDownload(<WebSocket>socket, path, is_dir);
+        if (socket) {
+          handleFileDownload(socket, path, is_dir);
+        }
       });
 
       mittBus.on(
         'file-manage',
         ({ path, type, new_name }: { path: string; type: ManageTypes; new_name?: string }) => {
+          if (!socket) return;
+
           switch (type) {
             case ManageTypes.CREATE: {
-              handleFileCreate(<WebSocket>socket, path);
+              handleFileCreate(socket, path);
               break;
             }
             case ManageTypes.CHANGE: {
-              handleChangePath(<WebSocket>socket, path);
+              handleChangePath(socket, path);
               break;
             }
             case ManageTypes.REFRESH: {
-              refresh(<WebSocket>socket, path);
+              refresh(socket, path);
               break;
             }
             case ManageTypes.RENAME: {
-              handleFileRename(<WebSocket>socket, path, new_name!);
+              handleFileRename(socket, path, new_name!);
               break;
             }
             case ManageTypes.REMOVE: {
-              handleFileRemove(<WebSocket>socket, path);
+              handleFileRemove(socket, path);
               break;
             }
           }
         }
       );
+
+      return {
+        cleanup,
+        socket
+      };
+    } catch (error) {
+      cleanup();
+      console.error('File management initialization failed:', error);
+      throw error;
     }
   }
 
-  init();
+  return init();
 };
 
 export const unloadListeners = () => {
```

Comment:
The code seems to be incomplete with just part of its implementation missing, which makes it difficult to verify or evaluate the potential issues accurately. It would help greatly if you can share the complete source code so that I could perform an in-depth analysis.

As always, remember while using raw binary content like this that your machine might not recognize it because some parts were cut off from the original document.